sdk/azcore/credential.go

```diff
@@ -44,3 +45,8 @@ type AccessToken struct {
 	ExpiresIn json.Number `json:"expires_in"`
```

Comment:
Shouldn't have both ExpiresIn and ExpiresOn, Only ExpiresOn

Reply:
I might change this if I see I do need to handle ExpiresIn differently for the MSI credential, but for Go this is a good solutions since we need to use ExpiresIn to unmarshal the JSON response (and the unmarshal needs to be done on an exported field). I'll look into the token responses for MSI and get back to this

Reply:
I do seem to recall that one type of endpoint only returned `ExpiresIn`, but if that's the case you should deserialize to an internal struct which has both fields and then calculate `ExpiresOn` from `ExpiresIn`. The AccessToken struct which is return from `TokenCredential.GetToken` should only have `ExpiresOn`

Reply:
Adding internal struct for marshaling purposes in next PR